File: Android/shared/src/main/kotlin/shared/dto/objects/Master.kt
@file:UseSerializers(UUIDSerializer::class, URISerializer::class)

package shared.dto.objects

import shared.serializers.URISerializer
import java.net.URI

import kotlinx.serialization.Serializable
import kotlinx.serialization.UseSerializers
import shared.dto.protocols.Responsable
import shared.serializers.UUIDSerializer
import java.util.*

object Master {
    data object Responses {
        @Serializable
        data class Salon(
            val id: UUID,
            val nickName: String,
            val logo: URI,
            val position: String,
        ) : Responsable
    }
}

File: iOS-legacy/DTOs/Objects/Customer/Master.swift
import Foundation

public enum Master {
    public enum Responses {}
}

public extension Master.Responses {
    typealias Masters = [Master]
    
    struct Master: Codable, Sendable {
        public let id: UUID
        public let nickname: String
        public let logo: URL
        public let position: String
        
        public init(id: UUID, nickname: String, logo: URL, position: String) {
            self.id = id
            self.nickname = nickname
            self.logo = logo
            self.position = position
        }
    }
}

